moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -799,6 +799,45 @@ bool moveit::core::RobotState::isValidVelocityMove(const RobotState& other, cons
         return false;
     }
   }
+  return true;
+}
+
+bool moveit::core::RobotState::isValidVelocityMove(const JointModelGroup* group,
+                                                   const std::vector<double>& from_joint_pose,
+                                                   const std::vector<double>& to_joint_pose, double dt) const
+{
+  // Check for equal sized arrays
+  if (from_joint_pose.size() != to_joint_pose.size())
+  {
+    logError("To and from joint poses are of different sizes.");
+    return false;
+  }
+
+  return isValidVelocityMove(group, &from_joint_pose[0], &to_joint_pose[0], from_joint_pose.size(), dt);
+}
+
+bool moveit::core::RobotState::isValidVelocityMove(const JointModelGroup* group, const double* from_joint_pose,
+                                                   const double* to_joint_pose, std::size_t array_size, double dt) const
+{
+  const std::vector<const JointModel::Bounds*>& bounds = group->getActiveJointModelsBounds();
+  const std::vector<unsigned int>& bij = group->getKinematicsSolverJointBijection();
+
+  for (std::size_t i = 0; i < array_size; ++i)
+  {
+    double dtheta = std::abs(from_joint_pose[i] - to_joint_pose[i]);
+    const std::vector<moveit::core::VariableBounds>* var_bounds = bounds[bij[i]];
+
+    if (var_bounds->size() != 1)
+    {
+      logError("Attempting to check velocity bounds for waypoint move with joints that have multiple variables");
+      return false;
+    }
+    const double max_velocity = (*var_bounds)[0].max_velocity_;
+
+    double max_dtheta = dt * max_velocity;
+    if (dtheta > max_dtheta)
+      return false;
```

Comment:
It's going to be less efficient, but would it be an idea to add a `ROS_DEBUG..` here with the joint that makes the check fail?

Reply:
ROS_DEBUG isn't used inside RobotState, but we could use ``logDebug``. However I agree that its less efficient so would prefer not.

Reply:
Done